File: source/Core/BSP/Magic/bl_mcu_sdk/examples/lvgl/demo/lv_ex_widgets/lv_ex_btnmatrix/lv_ex_btnmatrix_1.py
def event_handler(source,evt):
    if evt == lv.EVENT.VALUE_CHANGED:
        print("Toggled")
    txt = source.get_active_btn_text()
    print("%s was pressed"%txt)

# Create a screen and load it
scr = lv.obj()
lv.scr_load(scr)

btnm_map = ["1","2", "3", "4", "5", "\n",
            "6", "7", "8", "9", "0", "\n",
            "Action1", "Action2", ""]

btnm_ctlr_map = [46|lv.btnmatrix.CTRL.NO_REPEAT,46,46,46,46,
                 46,46,46,46,46,
                 115,115]

# create a button matrix
btnm1 = lv.btnmatrix(lv.scr_act(),None)
btnm1.set_map(btnm_map)
#btnm1.set_ctrl_map(btnm_ctlr_map)
btnm1.set_btn_ctrl(10,lv.btnmatrix.CTRL.CHECKABLE)
btnm1.set_btn_ctrl(11,lv.btnmatrix.CTRL.CHECK_STATE)
btnm1.set_width(230)
btnm1.align(None,lv.ALIGN.CENTER,0,0)
# attach the callback
btnm1.set_event_cb(event_handler)

File: source/Core/BSP/Magic/bl_mcu_sdk/examples/lvgl/demo/lv_ex_widgets/lv_ex_btnmatrix/lv_ex_btnmatrix_1.c
#include "../../../lv_examples.h"
#include "bflb_platform.h"
#if LV_USE_BTNMATRIX

static void event_handler(lv_obj_t *obj, lv_event_t event)
{
    if (event == LV_EVENT_VALUE_CHANGED) {
        const char *txt = lv_btnmatrix_get_active_btn_text(obj);
        MSG("%s was pressed\n", txt);
    }
}

static const char *btnm_map[] = { "1", "2", "3", "4", "5", "\n",
                                  "6", "7", "8", "9", "0", "\n",
                                  "Action1", "Action2", "" };

void lv_ex_btnmatrix_1(void)
{
    lv_obj_t *btnm1 = lv_btnmatrix_create(lv_scr_act(), NULL);
    lv_btnmatrix_set_map(btnm1, btnm_map);
    lv_btnmatrix_set_btn_width(btnm1, 10, 2); /*Make "Action1" twice as wide as "Action2"*/
    lv_btnmatrix_set_btn_ctrl(btnm1, 10, LV_BTNMATRIX_CTRL_CHECKABLE);
    lv_btnmatrix_set_btn_ctrl(btnm1, 11, LV_BTNMATRIX_CTRL_CHECK_STATE);
    lv_obj_align(btnm1, NULL, LV_ALIGN_CENTER, 0, 0);
    lv_obj_set_event_cb(btnm1, event_handler);
}

#endif
